File: javascript/rangeExtraction.js
function solution(list){
  let index = 1, result = '', storage = [list[0]]
  while(index < list.length) {
    if(list[index] === storage[storage.length - 1] + 1) { storage.push(list[index]) }
    else {
      if (storage.length >= 3) { result += `${storage[0]}-${storage[storage.length - 1]},` }
      else if (storage.length === 2) { result += `${storage[0]},${storage[1]},` }
      else { result += `${storage[0]},` }
      storage = [list[index]]
    }
    index++
   }
   if (storage.length >= 3) { result += `${storage[0]}-${storage[storage.length - 1]}` }
   else if (storage.length === 2) { result += `${storage[0]},${storage[1]}` }
   else { result += `${storage[0]}` }
   return result
 }

File: python/range_extraction.py
def solution(args):
    index, result, storage = 1, "", [args[0]]
    while index < len(args):
        if args[index] == storage[-1] + 1:
            storage.append(args[index])
        else:
            if len(storage) >= 3:
                result += f"{storage[0]}-{storage[-1]},"
            elif len(storage) == 2:
                result += f"{storage[0]},{storage[1]},"
            else:
                result += f"{storage[0]},"
            storage = [args[index]]
        index += 1
    if len(storage) >= 3:
        result += f"{storage[0]}-{storage[-1]}"
    elif len(storage) == 2:
        result += f"{storage[0]},{storage[1]}"
    else:
        result += f"{storage[0]}"
    return result